src/test/java/ru/hogwarts/school/HogwartsApplicationTests.java

```diff
@@ -1,13 +1,257 @@
+    }
+
+    @Test
+    public void testGetStudentById() {
+        int id = 8;
+        String name = "student3";
+        int age = 14;
+        Student student = new Student(id, name, age);
+
+        ResponseEntity<Student> response = this.restTemplate
+                .getForEntity("http://localhost:" + port + "/student/" + id, Student.class);
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+        assertThat(response.getBody()).isEqualTo(student);
+    }
+
+    @Test
+    public void removeStudentById() {
+        int id = 9;
+        String name = "student4";
+        int age = 16;
+        Student student = new Student(id, name, age);
+
+        ResponseEntity<Student> response = this.restTemplate
+                .getForEntity("http://localhost:" + port + "/student/" + id, Student.class);
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+        assertThat(response.getBody()).isEqualTo(student);
+
+        this.restTemplate.delete("http://localhost:" + port + "/student/" + id);
+
+        ResponseEntity<Student> response2 = this.restTemplate
+                .getForEntity("http://localhost:" + port + "/student/" + id, Student.class);
+        assertThat(response2.getStatusCode()).isEqualTo(HttpStatus.NO_CONTENT);
+    }
+
+    @Test
+    public void testEditStudent() {
+        int id = 10;
+        String name = "student5";
+        int age = 14;
+        Student studentBefore = new Student(id, name, age);
+
+        ResponseEntity<Student> response = this.restTemplate
+                .getForEntity("http://localhost:" + port + "/student/" + id, Student.class);
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+        assertThat(response.getBody()).isEqualTo(studentBefore);
+
+        int id2 = 10;
+        String name2 = "student5";
+        int age2 = 15;
+        Student studentAfter = new Student(id2, name2, age2);
+
+        this.restTemplate.put("http://localhost:" + port + "/student", studentAfter);
+
+        ResponseEntity<Student> response2 = this.restTemplate
+                .getForEntity("http://localhost:" + port + "/student/" + id, Student.class);
+        assertThat(response2.getStatusCode()).isEqualTo(HttpStatus.OK);
+        assertThat(response2.getBody()).isEqualTo(studentAfter);
+    }
+
+    @Test
+    public void testGetSameAgeStudents() {
+        int age = 15;
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Accept", "application/json");
+        headers.set("Authorization", "Bearer JWT TOKEN HERE");
+        HttpEntity<Object> requestEntity = new HttpEntity<>(null, headers);
+
+        ResponseEntity<Collection<Student>> response = restTemplate
+                .exchange("http://localhost:" + port + "/student/age",
+                HttpMethod.GET,
+                requestEntity,
+                new ParameterizedTypeReference<Collection<Student>>() {
+                }, age);
```

Comment:
```suggestion
        int age = 15;

        studentRepository.save(new Student(1,"name",15));

        HttpHeaders headers = new HttpHeaders();
        headers.set("Accept", "application/json");
        headers.set("Authorization", "Bearer JWT TOKEN HERE");
        HttpEntity<Object> requestEntity = new HttpEntity<>(null, headers);

        ResponseEntity<Collection<Student>> response = restTemplate
                .exchange("/student/age?age=" + age,
                HttpMethod.GET,
                requestEntity,
                new ParameterizedTypeReference<Collection<Student>>() {
                }, age);
```
Давай попробуем так:
То есть перед вызовом эндпоинта мы создадим студента
И немного поменяем url